src/marine/copernicus_l3swh2ioda.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python3
+
+#
+# (C) Copyright 2021 UCAR
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+
+import sys
+import argparse
+import netCDF4 as nc
+import numpy as np
+from datetime import datetime, timedelta
+import pytz
+import os
+from pathlib import Path
+
+IODA_CONV_PATH = Path(__file__).parent/"../lib/pyiodaconv"
+if not IODA_CONV_PATH.is_dir():
+    IODA_CONV_PATH = Path(__file__).parent/'..'/'lib-python'
+sys.path.append(str(IODA_CONV_PATH.resolve()))
+
+import ioda_conv_engines as iconv
+from collections import defaultdict, OrderedDict
+from orddicts import DefaultOrderedDict
+
+locationKeyList = [
+    ("latitude", "float"),
+    ("longitude", "float"),
+    ("datetime", "string"),
+]
+
+obsvars = {
+    'swh': 'swh',
+}
+
+AttrData = {
+    'converter': os.path.basename(__file__),
+    'nvars': np.int32(len(obsvars)),
+}
+
+DimDict = {
+}
+
+VarDims = {
+    'swh': ['nlocs'],
+}
+
+
+class copernicus(object):
+    def __init__(self, filename, factor):
+        # Create a simple data structure for Copernicus L3 swh near-realtime product obtained from
+        # ftp://nrt.cmems-du.eu/Core/WAVE_GLO_WAV_L3_SWH_NRT_OBSERVATIONS_014_001/
+        ncd = nc.Dataset(filename, 'r')
+        self.lons = np.squeeze(ncd.variables['longitude'][:]).ravel()
+        self.lats = np.squeeze(ncd.variables['latitude'][:]).ravel()
+        self.time = np.squeeze(ncd.variables['time'][:]).ravel()
+        self.swh = np.squeeze(ncd.variables['VAVH'][:]).ravel()
+        self.err = factor*self.swh
+        self.date = ncd.getncattr('first_meas_time')
+        ncd.close()
+
+        # set time stamp for all obs
+        self.datetime = np.empty_like(self.swh, dtype=object)
+        for t in range(len(self.datetime)):
+            self.datetime[t] = datetime.utcfromtimestamp(self.time[t]+datetime(2000, 1, 1, 0, tzinfo=pytz.UTC).timestamp()).strftime("%Y-%m-%dT%H:%M:%SZ")
+
+        # Remove observations out of sane bounds
+        qci = np.where(self.swh > 0.001)
```

Comment:
dumb questions, why are _swh_ obs between 0.0 and 0.001 bad?

Reply:
Good catch, @guillaumevernieres. This is just an arbitrary value I picked. Actually we should consider all swh obs > 0.0, especially these data has been pre-QC/validated against  wave buoy data. Changed this line to self.swh > 0.0.